content/blog/gain-platform-superpowers-with-the-kebap-stack/index.md

```diff
@@ -0,0 +1,295 @@
+the [Pulumi Provider](https://external-secrets.io/latest/provider/pulumi/) for ESO. This way, we can manage the secrets
+in the same way as the rest of the infrastructure,
+ensuring that the secrets are managed in a secure and consistent manner across different environments.
+
+To give the Pulumi Kubernetes Operator access to the secrets, we create following `ClusterSecretStore` object:
+
+```yaml
+apiVersion: external-secrets.io/v1beta1
+kind: ClusterSecretStore
+metadata:
+  name: secret-store
+spec:
+  provider:
+    pulumi:
+      organization: <org>
+      environment: <env>
+      accessToken:
+        secretRef:
+          name: pulumi-access-token
+          key: PULUMI_ACCESS_TOKEN
+          namespace: argocd
+```
+
+And then we can define the `ExternalSecret` with all the secrets that we want to manage and pass to our Operator as
+environment variables `extraEnv`:
+
+```yaml
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: pulumi-operator-secrets
+spec:
+  refreshInterval: 20s
+  secretStoreRef:
+    kind: ClusterSecretStore
+    name: secret-store
+  data:
+  - secretKey: pulumi-access-token
+    remoteRef:
+      key: "backstage.pulumi-pat"
+  - secretKey: do-token
+    remoteRef:
+      key: "backstage.do"
+```
+
+### Backstage
+
+<img src="b.png" width="500px">
+
+Backstage is an open platform for building developer portals. It provides a centralized place for managing software
+catalogs, documentation, and tooling. This layer helps in organizing microservices and infrastructure, streamlining the
+process for developers to create, manage, and explore services.
+
+We also installed the [Pulumi plugin](https://github.com/pulumi/pulumi-backstage-plugin) for Backstage, which allows us
+to manage our infrastructure as code directly from the Backstage UI. There are several ways to interact with the plugin.
+If you want to use the scaffolding feature, you can add following `step` for example to your `Template` definition:
+
+```yaml
+apiVersion: scaffolder.backstage.io/v1beta3
+kind: Template
+metadata:
+  name: kubernetes-template
+  title: Kubernetes Cluster
+  description: |
+    A template for creating a new Kubernetes Cluster.
+  tags:
+  - pulumi
+  - kubernetes
+spec:
+  steps:
+  - id: pulumi-new-component
+    name: Cookie cut the component Pulumi project
+    action: pulumi:new
+    input:
+      name: "${{ parameters.component_id }}-infrastructure"
+      description: ${{ parameters.description | dump }}
+      organization: ediri
+      stack: ${{ parameters.stack }}
+      template: "https://github.com/my-silly-organisation/microservice-civo/tree/main/infrastructure-${{ parameters.cloud }}-${{ parameters.language }}"
+      config:
+        "node:node_count": "${{ parameters.nodeCount }}"
+      folder: .
+```
+
+### Argo CD
+
+<img src="a.png" width="500px">
+
+Argo CD handles the deployment layer, ensuring that the current deployment matches what is declaratively defined in the
```

Comment:
I can't tell if the Argo layer is handling deployments or is the Pulumi K8s Operator handling the deployments within the K8s clusters.

Reply:
I am also having trouble understanding Backstage's relationship with Argo. 

I think you should probably explain it not in the KEBAP order but more in the workflow order. 

K8s is the container orch layer
Pulumi then manages the cloud infra that the K8s clusters runs in
All deployments to infra within the K8s cluster is done by the Pulumi K8s operators
Argo does X
Backstage interfaces with Pulumi to provide a GUI to templates which are executed by the K8s operator
ESO gives secrets...